log/src/main/java/org/apache/felix/log/LogServiceImpl.java

```diff
@@ -126,8 +125,8 @@ public Logger getLogger(String name) {
 
     @Override
     public Logger getLogger(Class<?> clazz) {
-        LogService logService = getLogService(clazz);
-        return logService.getLogger(clazz.getName());
+        Bundle bundle = FrameworkUtil.getBundle(clazz);
```

Comment:
The class here is only to get the name of the logger.  It is not intended to get a logger for another bundle.

Reply:
got it! fixing